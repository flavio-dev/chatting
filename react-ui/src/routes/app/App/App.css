.App, .AppNotTyped {
  transition: all 0.3s ease;
  background-color: #F4F9FF;
  color: #3A2424;
  font-size: 30px;
  padding: 40px 0;
  text-align: center; }
  @media only screen and (min-width: 601px) {
    .App, .AppNotTyped {
      padding: 60px 0; } }

.AppContent {
  margin-left: auto;
  margin-right: auto;
  width: 95%; }
  @media only screen and (min-width: 601px) {
    .AppContent {
      width: 90%; } }

.AppNotTyped {
  background-color: #3A2424;
  color: #F4F9FF; }
  .AppNotTyped .AppInput {
    border: 5px solid #F4F9FF; }
  .AppNotTyped .AppButton {
    border: 5px solid #00A0CC; }
    .AppNotTyped .AppButton:hover {
      border-color: #F4F9FF; }

.AppTitle {
  font-family: "Lobster", Helvetica, sans-serif;
  font-size: 40px;
  margin-bottom: 60px; }
  @media only screen and (min-width: 601px) {
    .AppTitle {
      font-size: 60px;
      margin-bottom: 40px; } }

.AppInput {
  transition: border 0.3s ease;
  font-family: "Lobster", Helvetica, sans-serif;
  background-color: #F4F9FF;
  border: 5px solid #3A2424;
  border-radius: 5px;
  color: #3A2424;
  padding: 20px;
  width: 80%; }
  .AppInput::placeholder {
    color: #B78686; }

.AppButtonContainer {
  display: flex;
  justify-content: center;
  margin: 60px 0; }
  @media only screen and (min-width: 601px) {
    .AppButtonContainer {
      margin: 40px 0; } }

.AppButton {
  transition: all 0.3s ease;
  background-color: #00A0CC;
  border: 5px solid #00A0CC;
  border-radius: 5px;
  color: #F4F9FF;
  display: block;
  font-family: "Lobster", Helvetica, sans-serif;
  padding: 20px; }
  .AppButton:hover {
    background-color: #F4F9FF;
    color: #00A0CC; }

.AppListUsers {
  margin-bottom: 20px; }
  .AppListUsers p {
    margin-bottom: 10px; }

.AppListUser {
  align-items: baseline;
  display: flex;
  justify-content: center; }

.AppOnline {
  background-color: #95FF87;
  border-radius: 10px;
  margin-right: 10px;
  height: 15px;
  width: 15px; }
