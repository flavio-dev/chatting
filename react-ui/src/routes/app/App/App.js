import React, { Component } from 'react'
import PropTypes from 'prop-types'

import styles from './App.css'

class App extends Component {
  constructor(props) {
    super(props)
    this.state = {
      users: [],
      numUsers: 0,
      userId: ''
    }

    this.interval = null

    this.typing = this.typing.bind(this)
    this.enterChat = this.enterChat.bind(this)
  }

  componentDidMount() {
    this.props.getInitialListUsers()
    this.startPolling()
  }

  componentWillUnmount() {
    this.clearPolling()
  }

  componentWillReceiveProps(nextProps) {
    this.setState({users: nextProps.users})
    this.setState({numUsers: nextProps.numUsers})
  }

  typing(event) {
    this.setState(
      {userId: event.target.value}
    )
  }

  enterChat() {
    const slug = this.state.userId.toLowerCase()
    this.props.redirectToUser(slug)
  }

  startPolling() {
    this.interval = setInterval(() => this.props.getInitialListUsers(), 5000);
  }

  clearPolling() {
    clearInterval(this.interval)
  }

  render() {
    const styleTyped = (this.state.userId.length)
      ? 'App'
      : 'AppNotTyped'
    return (
      <div className={styles[styleTyped]}>
        <div className={styles.AppContent}>
          <h2 className={styles.AppTitle}>Welcome to Chatting {this.state.userId}!</h2>
          <input
            className={styles.AppInput}
            type='text'
            value={this.state.userId}
            onChange={this.typing}
            placeholder='Enter your name' />
          <div className={styles.AppButtonContainer}>
            <button className={styles.AppButton} onClick={this.enterChat}>Enter the chatroom</button>
          </div>
          <p className={styles.AppListUsers}>There are currently {this.state.numUsers} user(s) online</p>
          {this.state.users.map((user, index) => {
            return (
              <div className={styles.AppListUser} key={index}>
                <div className={styles.AppOnline} /><span>{user}</span>
              </div>
            )
          })}
        </div>
      </div>
    )
  }
}

App.propTypes = {
  users: PropTypes.array,
  numUsers: PropTypes.number,
  getInitialListUsers: PropTypes.func,
  redirectToUser: PropTypes.func
}

export default App
