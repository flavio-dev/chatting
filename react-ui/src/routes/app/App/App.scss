@import "globals.scss";

.App {
  @include transitions(all);
  background-color: $color-white;
  color: $color-chocolate;
  font-size: 30px;
  padding: 40px 0;
  text-align: center;

  @media #{$medium-and-up} {
    padding: 60px 0;
  }
}

.AppContent {
  @include default-wrapper-width();
}

.AppNotTyped {
  @extend .App;
  background-color: $color-chocolate;
  color: $color-white;

  .AppInput {
    border: 5px solid $color-white;
  }

  .AppButton {
    border: 5px solid $color-blue-sky;

    &:hover {
      border-color: $color-white;
    }
  }
}

.AppTitle {
  font-family: $title-font-family;
  font-size: 40px;
  margin-bottom: 60px;

  @media #{$medium-and-up} {
    font-size: 60px;
    margin-bottom: 40px;
  }
}

.AppInput {
  @include transitions(border);
  font-family: $title-font-family;
  background-color: $color-white;
  border: 5px solid $color-chocolate;
  border-radius: 5px;
  color: $color-chocolate;
  padding: 20px;
  width: 80%;

  &::placeholder {
    color: $color-milk-chocolate;
  }
}

.AppButtonContainer {
  display: flex;
  justify-content: center;
  margin: 60px 0;

  @media #{$medium-and-up} {
    margin: 40px 0;
  }
}

.AppButton {
  @include transitions(all);
  background-color: $color-blue-sky;
  border: 5px solid $color-blue-sky;
  border-radius: 5px;
  color: $color-white;
  display: block;
  font-family: $title-font-family;
  padding: 20px;

  &:hover {
    background-color: $color-white;
    color: $color-blue-sky;
  }
}

.AppListUsers {
  margin-bottom: 20px;

  p {
    margin-bottom: 10px;
  }
}

.AppListUser {
  align-items: baseline;
  display: flex;
  justify-content: center;
}

.AppOnline {
  @include online();
}
