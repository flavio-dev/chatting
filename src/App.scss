.App {
  text-align: center;
}

.AppLogo {
  height: 80px;
  animation: App-logo-spin infinite 20s linear;
}

.AppHeader {
  background-color: #262;
  height: 150px;
  padding: 20px;
  color: white;
}

.AppIntro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
