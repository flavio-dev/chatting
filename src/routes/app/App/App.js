import React, { Component } from 'react'
import PropTypes from 'prop-types'

import styles from './App.css'

class App extends Component {
  constructor(props) {
    super(props)
    this.state = {
      users: [],
      numUsers: 0,
      userId: '',
      hasTyped: false
    }

    this.typing = this.typing.bind(this)
    this.enterChat = this.enterChat.bind(this)
  }

  componentDidMount() {
    this.props.getInitialListUsers()
  }

  componentWillReceiveProps(nextProps) {
    this.setState({users: nextProps.users})
    this.setState({numUsers: nextProps.numUsers})
  }

  typing(event) {
    this.setState(
      {userId: event.target.value}
    )

    if (event.target.value && event.target.value.length > 0) {
      this.setState({
        hasTyped: true
      })
    } else {
      this.setState({
        hasTyped: false
      })
    }
  }

  enterChat() {
    const slug = this.state.userId.toLowerCase()
    this.props.redirectToUser(slug)
  }

  render() {
    let style = this.state.hasTyped ? 'App' : 'AppNotTyped'
    return (
      <div className={styles[style]}>
        <div className={styles.AppContent}>
          <h2 className={styles.AppTitle}>Welcome to Chatting {this.state.userId}!</h2>
          <input
            className={styles.AppInput}
            type='text'
            value={this.state.userId}
            onChange={this.typing}
            placeholder='Enter your name' />
          <div className={styles.AppButtonContainer}>
            <button className={styles.AppButton} onClick={this.enterChat}>Enter the chatroom</button>
          </div>
          <p className={styles.AppListUsers}>There are currently {this.state.numUsers} user(s) online</p>
          {this.state.users.map((user, index) => {
            return (
              <div className={styles.AppListUser} key={index}>
                <div className={styles.AppOnline} /><span>{user}</span>
              </div>
            )
          })}
        </div>
      </div>
    )
  }
}

App.propTypes = {
  users: PropTypes.array,
  numUsers: PropTypes.number,
  getInitialListUsers: PropTypes.func,
  redirectToUser: PropTypes.func
}

export default App
