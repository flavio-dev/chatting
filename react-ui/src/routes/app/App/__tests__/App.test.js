import React from 'react'
import App from '../App'
import { shallow, mount, setState } from 'enzyme'
import renderer from 'react-test-renderer'

describe('Testing App.js', () => {
  it('renders correctly with empty list of users', () => {
    const getInitialListUsers = jest.fn(() => [])
    const redirectToUser = jest.fn()
    const tree = renderer.create(
      <App
        user={[]}
        numUsers={0}
        getInitialListUsers={getInitialListUsers}
        redirectToUser={redirectToUser}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with some users already connected', () => {
    const getInitialListUsers = jest.fn()
    const redirectToUser = jest.fn()
    const tree = renderer.create(
      <App
        user={['user']}
        numUsers={1}
        getInitialListUsers={getInitialListUsers}
        redirectToUser={redirectToUser}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('changes the class when starting to type', () => {
    const getInitialListUsers = jest.fn()
    const redirectToUser = jest.fn()
    const wrapper = shallow(
      <App
        user={[]}
        numUsers={0}
        getInitialListUsers={getInitialListUsers}
        redirectToUser={redirectToUser}
      />
    )
    expect(wrapper.find('.AppNotTyped').length).toBe(1)
    expect(wrapper.find('.App').length).toBe(0)
    wrapper.setState({userId: 'text'})
    expect(wrapper.find('.AppNotTyped').length).toBe(0)
    expect(wrapper.find('.App').length).toBe(1)
  })
})
