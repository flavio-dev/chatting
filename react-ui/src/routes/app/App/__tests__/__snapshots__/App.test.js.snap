// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App.js renders correctly with empty list of users 1`] = `
<div
  className="AppNotTyped"
>
  <div
    className="AppContent"
  >
    <h2
      className="AppTitle"
    >
      Welcome to Chatting 
      
      !
    </h2>
    <input
      className="AppInput"
      onChange={[Function]}
      placeholder="Enter your name"
      type="text"
      value=""
    />
    <div
      className="AppButtonContainer"
    >
      <button
        className="AppButton"
        onClick={[Function]}
      >
        Enter the chatroom
      </button>
    </div>
    <p
      className="AppListUsers"
    >
      There are currently 
      0
       user(s) online
    </p>
  </div>
</div>
`;

exports[`Testing App.js renders correctly with some users already connected 1`] = `
<div
  className="AppNotTyped"
>
  <div
    className="AppContent"
  >
    <h2
      className="AppTitle"
    >
      Welcome to Chatting 
      
      !
    </h2>
    <input
      className="AppInput"
      onChange={[Function]}
      placeholder="Enter your name"
      type="text"
      value=""
    />
    <div
      className="AppButtonContainer"
    >
      <button
        className="AppButton"
        onClick={[Function]}
      >
        Enter the chatroom
      </button>
    </div>
    <p
      className="AppListUsers"
    >
      There are currently 
      0
       user(s) online
    </p>
  </div>
</div>
`;
